### PR TITLE
Add some docs for the privacy notice templates #14233

### PR DESCRIPTION
**Pull request by stephaniehobson**
## One-line summary

Add some docs for the privacy notice templates.

## Significant changes and points to review

- added file
- added link to file

## Issue / Bugzilla link

This is to help with #14233

## Testing

`make livedocs` in your venv

### CODE REVIEW

**Review by janbrasna — COMMENTED** [review]
Nits: one typo and the rest is just formatting/Qs:

File: docs/legal-docs.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    ### This will be displayed as an h3 heading and will serve as a toggle for all its siblings until the next h3 {: provide-an-id }
+
+    * This will be a bulleted list that is collapsed under the h3 heading
+
+    This will be a paragraph that is collapsed under the h3 heading
+
+    ### Footnote {: footnote }
+
+    If a section contains a heading with an ID of `footnote` the section will be extracted and output as the last thing on the page.
```
[the remark this answers is not in the document]
Comment (stephaniehobson) [reply docs/legal-docs.rst:83]: Any heading.

**Comment by stephaniehobson** [comment]
Committed all of @janbrasna's excellent review suggestions. I'll look for someone to hit the green button.

**Review by craigcook — APPROVED** [review]
📜

**Pull request merged**